### PR TITLE
magento/magento2#22317: CodeSniffer should not mark correctly aligned DocBlock elements as code style violation.

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Fixed Issues (if relevant)

1. magento/magento2#22317: CodeSniffer should not mark correctly aligned DocBlock elements as code style violation.

### Manual testing scenarios (*)

1. Edit any Magento 2 class - align method params like shown [here](https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#format-consistency)
2. Run static tests.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dev/tests/static/framework/Magento/Sniffs/Annotation/MethodArgumentsSniff.php [hunk 1]
```diff
@@ -584,4 +594,80 @@ public function process(File $phpcsFile, $stackPtr)
...
+
+    /**
+     * Validates function params format consistency.
+     *
+     * @param array $paramDefinitions
+     * @param array $methodArguments
+     * @param File $phpcsFile
+     * @param array $paramPointers
+     *
+     * @see https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#format-consistency
```
Comment (orlangur) [inline dev/tests/static/framework/Magento/Sniffs/Annotation/MethodArgumentsSniff.php:608]: Omg, didn't even know we have such a dumb rule O.o

**Comment by orlangur** [comment]
@p-bystritsky did you align such changes with @lenaorobei or @paliarush? I believe there should be some plan elaborated in regards to annotation sniffs as current ones are pretty lame, probably there is no need to try improve them sporadically.

**Comment by lenaorobei** [comment]
Hi @p-bystritsky! The idea is to implement proper DocBlock checking sniffs and remove Annotations sniffs from Magento at all. Here are related tasks: https://github.com/magento/magento-coding-standard/issues/58 and https://github.com/magento/magento-coding-standard/issues/54

I [did not merge those Annotations](https://github.com/magento/magento2/blob/2.3-develop/dev/tests/static/framework/Magento/ruleset.xml#L12) sniffs to the [consolidated Coding Standard](https://github.com/magento/magento-coding-standard), because they produce false-positive findings.

**Comment by sivaschenko** [comment]
@lenaorobei should we merge this pull request to improve the existing sniffs until the DocBlock checking is in progress?

**Comment by lenaorobei** [comment]
@sivaschenko I don't see any reasons to not to merge, since it improves developer experience.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4774](https://jira.corp.magento.com/browse/ENGCOM-4774) has been created to process this Pull Request

**Pull request merged**